buidl/message.py

```diff
@@ -0,0 +1,224 @@
+#    vout[0].scriptPubKey = OP_RETURN
+#
+def create_to_sign_tx(to_spend_tx_hash, sig_bytes=None):
+
+    if (sig_bytes and is_full_signature(sig_bytes)):
+        
+        sig_stream = io.BytesIO(sig_bytes)
+        to_sign = Tx.parse(sig_stream)
+        
+        if (len(to_sign.tx_ins) > 1):
+            # TODO: Implement Proof of Funds
+            raise NotImplemented("Not yet implemented proof of funds yet")
+        elif (len(to_sign.tx_ins) == 0):
+            raise ValueError("No transaction input")
+        elif (to_sign.tx_ins[0].prev_tx != to_spend_tx_hash):
+            raise ValueError("The to_sign transaction input's prevtx id does not equal the calculated to_spend transaction id")
+        elif (len(to_sign.tx_outs) != 1):
+            raise ValueError("to_sign does not have a single TxOutput")
+        elif (to_sign.tx_outs[0].amount != 0):
+            raise ValueError("Value is Non 0", to_sign.tx_outs[0].amount)
+        elif(to_sign.tx_outs[0].script_pubkey.commands != [106]):
+            raise ValueError("ScriptPubKey incorrect", to_sign.tx_outs[0].script_pubkey)
+        else:
+            return to_sign
+            
+    else:
+        # signature is either None or an encoded witness stack
+        
+        # Identifies the index of the output from the virtual to_spend tx to be "spent"
+        prevout_index = 0
+        sequence = 0
+        # TxInput identifies the single output from the to_spend Tx
+        tx_input = TxIn(to_spend_tx_hash,prevout_index,script_sig=None,sequence=sequence)
+    
+        value = 0
+        # OP Code 106 for OP_RETURN
+        commands = [106]
+        scriptPubKey = Script(commands)
+
+        tx_output = TxOut(value,scriptPubKey)
+        locktime=0
+        version=0
+        tx_inputs = [tx_input]
+        tx_outputs = [tx_output]
+        network="mainnet"
+        # is to_sign always a segwit?
+        segwit=True
+    
+        # create unsigned to_sign transaction
+        to_sign_tx = Tx(version,tx_inputs,tx_outputs,locktime,network,segwit)
+        
+        if sig_bytes:
+            try:
+                stream = io.BytesIO(sig_bytes)
+                witness = Witness.parse(stream)
+                # Set the witness on the to_sign tx input
+                to_sign_tx.tx_ins[0].witness = witness
+            except:
+                # TODO: Fall back to legacy ...
+                print("Signature is niether an encoded witness or full transaction. Fall back to legacy")
+                return None
+                
+        return to_sign_tx
+
+# Test if sig_bytes can be decoded to a transaction
+# TODO: Is there a better way to test than this?
+def is_full_signature(sig_bytes):
+    try:
+        sig_stream = io.BytesIO(sig_bytes)
+
+        Tx.parse(sig_stream)
+    # TODO: more specific exception handling
+    except:
+        return False
+    return True
+        
+
+def sign_message(format: MessageSignatureFormat, private_key: PrivateKey, address: str, message: str):
```

Comment:
Nothing else in the library uses Type hints like this. I would prefer you specify in the documentation.
Also, please make clear what you are getting back when you call this function. What sort of signature is it? How do I verify it? What's the signature encoded in?

Reply:
I have removed type hints and added documentation in comments above the function